Makefile

```diff
@@ -1,31 +1,52 @@
+# Unless set otherwise, the container runtime is Docker
+DOCKER ?= docker
+
+prog ?= configurator
+git_tag := $(git describe --abbrev=0 --tags)
+sources := main.go $(wildcard cmd/*.go)
+plugin_source_prefix := pkg/generator/plugins
+plugin_sources := $(filter-out %_test.go,$(wildcard $(plugin_source_prefix)/*/*.go))
+plugin_binaries := $(addprefix lib/,$(patsubst %.go,%.so,$(notdir $(plugin_sources))))
 
 # build everything at once
+.PHONY: all
 all: plugins exe test
 
 # build the main executable to make configs
+.PHONY: main driver binaries exe
 main: exe
 driver: exe
-exe:
-	go build --tags=all -o configurator
+binaries: exe
+exe: $(prog)
+
+# build named executable from go sources
+$(prog): $(sources)
+    go build --tags=all -o $(prog)
```

Comment:
I think you have spaces here instead of a tab. Getting:
```
Makefile:24: *** missing separator.  Stop.
```

Reply:
Copy-pasta fail, I guess.